File: LeetCode/Easy/0009-palindrome-number/0009-palindrome-number.go
func isPalindrome(x int) bool {
    var arr []int
    num := x
    if (num < 0) {
        return false;
    }
    for num !=0 {
        arr = append(arr, num % 10)
        num = num / 10
    }
    fac := 1
    rev := 0
    for i := len(arr) -1; i>=0; i-- {
        rev += fac * arr[i]
        fac *= 10;
    }
    return x == rev
}

File: LeetCode/Easy/0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        int src = x;
        int rev = 0;

        while (x != 0) {
            rev *= 10;
            rev += x % 10;
            x = x/10;
        }
        return src == rev;
    }
}